### PR TITLE
fix(secondary-nav): mobile menu overflow

### LINKED ISSUE #557: [bug] rh-secondary-nav (!) Circular dependency
### Describe the bug

While using rollup to bundle `rh-secondary-nav` I see the following warning / error:

```
(!) Circular dependency
node_modules/@rhds/elements/elements/rh-secondary-nav/rh-secondary-nav-dropdown.js -> node_modules/@rhds/elements/elements/rh-secondary-nav/rh-secondary-nav-menu.js -> node_modules/@rhds/elements/elements/rh-secondary-nav/rh-secondary-nav-dropdown.js
```

It looks like `rh-secondary-nav-dropdown` is importing `rh-secondary-nav-menu` which is importing `rh-secondary-nav-dropdown`.

Looking into this with Steven it seems that `rh-secondary-nav-menu` is importing `rh-secondary-nav-dropdown` in order to define `isDropdown` _but_ `isDropdown` isn't currently being used.

So removing `isDropdown` within `rh-secondary-nav-dropdown` and removing the `rh-secondary-nav-menu` import resolves the *Circular dependency* issue. 

### Which Component?

rh-secondary-nav

### Steps to reproduce

1. Ask Zack to see the rollup config in rhdc.
2. Use that config to import rh-secondary-nav and check the error.

### Expected behaviour

There shouldn't be any circular dependencies if we can help it.

### Screenshots

_No response_

### Element HTML

_No response_

### Operating System (OS)

macOS

### Browser

_No response_

### More browser information

_No response_

### Additional Info

_No response_

### LINKED ISSUE #559: [bug] <secondary-nav> mobile nav menu height overflow
### Describe the bug

The mobile nav menu on small height screens does not overflow correctly when a long dropdown is present.  The cta button becomes inaccessible. when a dropdown is open.

### Which Component?

rh-secondary-nav

### Steps to reproduce

Go to the [Secondary Nav Demo](https://ux.redhat.com/components/secondary-nav/demo/dark-variant/) for the Dark variant
Set the viewport with height < 850px such as an iPhone.
Open the mobile menu.
Click "Other examples".
Try and scroll to CTA.

### Expected behaviour

The overflow should exist on the container not the nav list allowing the CTA to be scrolled into view.

### Screenshots

image

### Element HTML

_No response_

### Operating System (OS)

iOS, macOS

### Browser

Chrome, Safari, Safari iOS, Microsoft Edge, Firefox

### More browser information

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1.  Moved overflow mobile menu list to an outer container.
2. Removed orphaned code.
3. Closes #559 
4. Closes #557

## Testing Instructions

1. Go to the Secondary Nav Deploy Preview for the [Dark variant](https://deploy-preview-561--red-hat-design-system.netlify.app/components/secondary-nav/demo/dark-variant/) and the [Default](https://deploy-preview-561--red-hat-design-system.netlify.app/components/secondary-nav/demo/)
2. Set the viewport with height < 850px such as an iPhone.
5. Open the mobile menu.
6. Click "Other examples".
7. Try and scroll to CTA.  It should be visible

## Notes to Reviewers

You should be able to scroll to and view the CTA when the mobile menu is open when viewing on netlify deploy preview.

However, if you run the code locally in the dev demo `npm start` you will have to remove the demo's overflows in order to see the change correctly due to conflicting CSS.

### CODE REVIEW

File: elements/rh-secondary-nav/rh-secondary-nav-menu.css [hunk 1]
```diff
@@ -43,14 +43,16 @@
     position: absolute;
     left: 0;
     right: 0;
-    top: var(--_nav-height);
+    /* top: var(--_nav-height); */
```
Comment (heyMP) [inline elements/rh-secondary-nav/rh-secondary-nav-menu.css:46]: @zeroedin are you good with remove this?

**Comment by heyMP** [comment]
@zeroedin looks good.  Just one small question.

File: elements/rh-secondary-nav/rh-secondary-nav-menu.css (hunk 1 above)
Comment (zeroedin) [reply elements/rh-secondary-nav/rh-secondary-nav-menu.css]: Fixed in 007da26350aa49ff05695ad3de41c8f5145a7181

**Review by heyMP — APPROVED** [review]
LGTM 🔥

**Pull request merged**